accord-core/src/main/java/accord/messages/CheckStatus.java

```diff
@@ -145,14 +151,20 @@ public CheckStatusReply apply(SafeCommandStore safeStore)
         Ranges truncated = command.hasBeen(Truncated) ? safeStore.ranges().allBetween(command.txnId().epoch(), command.executeAtOrTxnId().epoch()) : Ranges.EMPTY;
         Status invalidIfNotAtLeast = invalidIfNotAtLeast(safeStore);
         boolean isCoordinating = isCoordinating(node, command);
+        Durability durability = command.durability();
+        Route<?> route = command.route();
+        if (Route.isFullRoute(route))
+            durability = Durability.merge(durability, safeStore.commandStore().durableBefore().min(txnId, route));
```

Comment:
`DurableBefore` only tells you that the covered range is durable, not that every transaction intersecting with that range is durable. So if you don't know the full route, you may only check part of the transaction is durable rather than the whole transaction.